docs/team/cowsaysbaa.adoc

```diff
@@ -0,0 +1,75 @@
+
+.Graphical user interface of ModuleBook
+image::ModuleBookUILayout.png[]
+
+ModuleBook is a desktop module manager application designed with the needs of NUS students in mind, to streamline the process of managing time and resources related to the modules offered by NUS. The user interacts with it using a CLI, and it has a GUI created with JavaFX. It is written in Java.
+
+As of the current build (v1.4), ModuleBook offers the following features:
+
+. Detailed archive of all modules offered by NUS (adaped from NUSMods API).
+. Track modules of interest.
+. Search for specific modules using search filters.
+. Add personalised deadlines to tracked modules.
+. Add links to tracked modules.
+
+My role was to adapt and migrate the base code of addressbook to the requirements of ModuleBook, as well as to design and enhance the codes for operations related to tracking a module. This includes the add, delete, list and find command.
+The following sections illustrate these enhancements in more detail, as well as the relevant documentation I have added to the user and developer guides in relation to these enhancements.
+
+== Summary of contributions
+
+This section documents my contribution to this team project in the from of code, documentation as well as supporting the team.
+
+* *Major enhancement*: added *the ability to find specific modules using filtered search.*
+** What it does: Allows the user to search for modules by filtering any combination of the following categories:
+. module
+. title
+. description
+. prerequsite modules
+. precluded modules
+. semesters the modules are offered
+** Justification: This feature should be very useful for freshmen students, or students who are unfamiliar with the large selection of modules offered by NUS. These students would be able to able to find a module of their interest without requiring in depth knowledge of specific details related to these modules.
+** Highlights: The search is is able to detect and ignore minor spelling mistakes when searching within the title or description. Searching is also extremely fast, with most search cases taking less than a second. These would definitely improve the user friendliness of the search feature.
+** Credits: Module information is directly adapted from the API provided by NUSMods. The adaptation and storage of module information is done by Ge Shuming, one of the students involved in this project. Without these information, this feature would not be possible.
+
+* *Minor enhancement*: improved the base add and delete module commands to be more in line with the program specifics.
+
+* *Code contributed*: [https://github.com/AY1920S1-CS2103T-W13-1/main/blob/master/src/main/java/seedu/module/logic/commands/FindCommand.java[Functional code]] [https://github.com/AY1920S1-CS2103T-W13-1/main/blob/master/src/test/java/seedu/module/logic/parser/FindCommandParserTest.java[Test code]]
```

Comment:
Should link to the reposense code contribution. https://nus-cs2103-ay1920s1.github.io/website/admin/project-deliverables.html#ppp-sections-to-include

Reply:
Thanks for the note. Fixed.